### PR TITLE
Improves the spam filter, moves variables to config

### PR DESCRIPTION
**Pull request by Rukofamicom**
## About The Pull Request
* Changes automute check into a rolling timer instead of being reset after every message
* Moves all of the fine-tuning into a config file where malicious parties are unable to see exact values without testing the filter directly

## Why It's Good For The Game
A filter that is easily bypassed by malicious parties but annoying to regular players is bad. This makes the filter more effective against malicious parties by making a rolling counter instead of a "pause a few seconds between spam bursts". Hiding the exact values from said malicious parties will also make it hard to tailor a macro to bypass it without trial and error. 

## Testing Photographs and Procedure

<details>
<summary>Debug text only exists in the video, not in this PR:</summary>

https://user-images.githubusercontent.com/9547572/170147805-a27a39cf-95e7-4c15-89d3-fddd0a8fd762.mp4

</details>

## Changelog
:cl: MNarath1, Ruko
config: Automute settings are now all defined via config to hide their exact values
admin: Automute filter is now better at stopping spam and worse at punishing innocent players who just type fast
/:cl:

### CODE REVIEW

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -151,20 +151,31 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
...
-	if(total_message_count >= SPAM_TRIGGER_AUTOMUTE)
-		to_chat(src, "<span class='userdanger'>You have exceeded the spam filter limit for too many messages. An auto-mute was applied. Make an adminhelp ticket if you think this was in error.</span>")
-		cmd_admin_mute(src, mute_type, TRUE)
-		return TRUE
+	if(COOLDOWN_TIMELEFT(src, total_count_reset) >= 30 SECONDS) //You have sent more messages in the past 30 seconds than is allowed by the spam filter
+		if(!COOLDOWN_FINISHED(src, warning_message_cooldown)) //ensures the player is warned before they are muted, since it is possible for some configurations to bypass the warning message. 
+			if(prob(95))
+				to_chat(src, "<span class='userdanger'>You have exceeded the spam filter limit. \n Make an adminhelp ticket if you think this was in error.</span>")
+			else
+				to_chat(src, "<span class='userdanger'>You have been 1984'd by the spam filter. \n Make an adminhelp ticket if you think this was in error.</span>")
```
Comment (PowerfulBacon) [inline code/modules/client/client_procs.dm]: Don't add memes into administrative messages

Comment (Rukofamicom) [reply code/modules/client/client_procs.dm]: rip. Worth a shot.

**Comment by PowerfulBacon** [comment]
You talk very slowly in the video and hit the spam filter anyway

**Comment by PowerfulBacon** [comment]
Im merging #6937 in favour for now, however I like the idea of the repeated words stacking. This PR is just too strict though.

**Comment by Rukofamicom** [comment]
@PowerfulBacon

I hit the spam filter because I repeated the same line twice. 
image
see the x2 next to where I say "5" and how it skipped the counter from 16 to 26? The first 5 pushed 16 from 12, and the second to 26. 

This video was also taken with fairly strict settings to showcase the counting. 
The configuration can be easily tweaked to allow any rate you want, including only making it really count for repeated messages again.

**Comment by Rukofamicom** [comment]
Just suggest a configuration and I can demonstrate it:

```
## Maximum rate of messages per minute before auto-mute kicks in. 
SPAM_TRIGGER_AUTOMUTE 30

## If a player sends an identical message twice or more times in a row, how many messages does it count as to the spam filter
SPAM_TRIGGER_IDENTICAL 5
```

This config for example: 
* Every unique message gets a 2 second counter
* Every repeat message gets a 10 second counter
* Counter goes down by 1 ever second. 
* If your counter hits 30, you get muted.

**Comment by PowerfulBacon** [comment]
> Every unique message gets a 2 second counter

I still feel like its quite strict with only allowing 1 message every 2 seconds, but we will have to see how it plays out.

**Comment by Rukofamicom** [comment]
Closing this for now but I'll keep the branch around for a while. 
Ping me in discord if you want to draw my attention back to it.

**Pull request closed without merging**